poc/vdaf_poc/test_utils.py

```diff
@@ -69,18 +71,88 @@ def run_vdaf_test(
 
 
+class VdafTestVectorOperationDict(TypedDict):
+    """
+    A description of one of the operations that should be performed with this
+    test vector.
+
+    Attributes:
+        operation: The type of operation to be performed. This is one of
+            "shard", "prep_init", "prep_shares_to_prep", "prep_next",
+            "aggregate", or "unshard".
+        round: The round number of the operation to be performed. This
+            determines which prepare share, prepare state, and/or prepare
+            message to use.
+        aggregator_id: The aggregator ID to use when performing this
+            operation. This determines which messages and which prepare state
+            to use, in addition to the aggregator ID argument itself.
+        report_index: The index of the report on which to perform this
+            operation. This is an index into the `prep` array.
+        success: If this is true, the operation should succeed, and its output
+            should match the corresponding values in the test vector. If this
+            is false, the operation should fail.
+    """
+    operation: (
+        Literal["shard"] | Literal["prep_init"]
+        | Literal["prep_shares_to_prep"] | Literal["prep_next"]
+        | Literal["aggregate"] | Literal["unshard"]
+    )
+    round: NotRequired[int]
+    aggregator_id: NotRequired[int]
+    report_index: NotRequired[int]
+    success: bool
+
+
 class VdafPrepTestVectorDict(Generic[Measurement], TypedDict):
+    """
+    This lists VDAF messages related to one report, from sharding to
+    preparation.
+
+    All VDAF messages are encoded to byte strings, and then hex-encoded.
+
+    Attributes:
+        measurement: The measurement used to produce the report.
+        nonce: The report's nonce.
+        rand: The randomness consumed by the sharding algorithm.
+        public_share: The public share from the report.
+        input_shares: The input shares from the report.
+        prep_shares: The prepare shares produced during preparation. This is
+            indexed first by round, and then by aggregator ID.
+        prep_messages: The prepare messages produced during aggregation. This
+            is indexed by round.
+        out_shares: The output shares produced by preparing this report. This
+            is indexed by round.
+    """
     measurement: Measurement
     nonce: str
+    rand: str
+    public_share: str
     input_shares: list[str]
     prep_shares: list[list[str]]
     prep_messages: list[str]
     out_shares: list[list[str]]
-    rand: str
-    public_share: str
 
 
 class VdafTestVectorDict(Generic[Measurement, AggParam, AggResult], TypedDict):
+    """
+    A test vector for evaluation of a VDAF on one or more reports.
+
+    All VDAF messages are encoded to byte strings, and then hex-encoded.
+
+    Attributes:
+        operations: A list of operations that should be executed using
+            messages from this test vector.
```

Comment:
A couple things we should clarify:

1. The order of operations: run the commands from `operations[0]` to `operations[-1]`.
2. When an input is needed from a previous operation. For example, to run `prep_next()`, you need to know the aggregator's prep state, which is not included in the test vector. We might resolve this by combining all of the "prep_*" commands into a single command "prep" like we do for "aggregate".
3. Whether there are "illegal" programs. For instance, is it valid to have an operation following one that fails to succeed?

order of operations is unclear
some "programs" are illegal (no operations allowed after failure)